src/Microsoft.AspNet.Diagnostics/ErrorPageMiddleware.cs

```diff
@@ -109,7 +115,7 @@ private Task DisplayException(HttpContext context, Exception ex)
                         Function = string.Empty
                     };
 
-                    ReadFrameContent(frame, fileContent, item.StartLine, item.EndLine);
+                    GetErrorCodeBlock(frame, fileContent, item.StartLine, item.EndLine);
```

Comment:
Could also make this method internal and build your own compilation exceptions to simulate :smile: